test/libyul/yulOptimizerTests/unusedStoreEliminator/function_side_effects_3.yul

```diff
@@ -0,0 +1,28 @@
+{
+    function conditionallyStop() {
+        if calldataload(0) { leave }
+        return(0, 0)
+    }
+    let x := 0
+    let y := 1
+    sstore(x, y) // used to be removed due to a bug
```

Comment:
Could you also add an opposite case that shows that the line does get removed when the termination is unconditional?

Reply:
It only *should* get removed, if the function unconditionally *doesn't* return... I can add that as test, though...

Reply:
Ah right. ok then, probably not that important. We have many test cases already.